src/hdmf/utils.py

```diff
@@ -116,7 +117,8 @@ def __format_type(argtype):
         raise ValueError("argtype must be a type, str, list, or tuple")
 
 
-def __parse_args(validator, args, kwargs, enforce_type=True, enforce_shape=True, allow_extra=False):   # noqa: C901
+def __parse_args(validator, args, kwargs, enforce_type=True, enforce_shape=True, allow_extra=False,  # noqa: C901
+                 allow_positional=True):
```

Comment:
If you want it configurable whether ``allow_positional`` should warn or raise and error, then you may want to make this an integer option (0=off, 1=warn, 2=error, 3="I'll call your parents")

Reply:
This is implemented now